### PR TITLE
feat(UserAgent): introduce addAlgoliaAgent method

### PR DESCRIPTION
**Pull request by tkrugg**
This introduces a addAlgoliaAgent method that can be also found in the
search client.

```ts
aa('addAlgoliaAgent', 'a random string')
```

### CODE REVIEW

File: lib/insights.ts [hunk 1]
```diff
@@ -66,6 +68,10 @@ class AlgoliaAnalytics {
   _userHasOptedOut: boolean;
   _cookieDuration: number;
 
+  // user agent
+  _ua: string = "";
```
Comment (tkrugg) [inline lib/insights.ts]: `_ua`  is the current name in the search client too
https://github.com/algolia/algoliasearch-client-javascript/blob/ead86654beb7bc5215bb8ff66d4c951637190a10/src/AlgoliaSearchCore.js#L175

File: lib/_algoliaAgent.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { version } from "../package.json";
+
+export const DEFAULT_ALGOLIA_AGENT = `Algolia insights for JavaScript (${version})`;
+
+export function addAlgoliaAgent(algoliaAgent) {
+  if (this._ua.indexOf(`;${algoliaAgent}`) === -1) {
+    this._ua += `;${algoliaAgent}`;
```
Comment (Haroenv) [inline lib/_algoliaAgent.ts]: @samouss separated this with spaces in the client

Comment (tkrugg) [reply lib/_algoliaAgent.ts]: I look for it as I wasn't sure where the spaces were but couldn't find it.
https://github.com/algolia/algoliasearch-client-javascript/blob/develop/src/AlgoliaSearchCore.js#L173-L177
Am I looking at the right branch?

Comment (Haroenv) [reply lib/_algoliaAgent.ts]: I think it’s in Robert’s branch still

Comment (tkrugg) [reply lib/_algoliaAgent.ts]: thanks! it's fixed here: 2546385

**Pull request merged**